test/grouping.jl

```diff
@@ -2691,4 +2691,122 @@ end
+        @test combine(gdf, :x => fun => :y) ≅ combine(gdf, :x => (x -> fun(x)) => :y)
+    end
+
+    for fun in (maximum, minimum, maximum∘skipmissing, minimum∘skipmissing),
+        col in ([1, 2, 3], [big(1.5), big(2.5), big(3.5)],
+                [true, false, true], [pi, pi, pi], [1//2, 1//3, 1//4],
+                Real[1, 1.5, 1//2], Number[1, 1.5, 1//2], Any[1, 1.5, 1//2],
+                [1, 2, missing], [big(1.5), big(2.5), missing],
+                [true, false, missing], [pi, pi, missing], [1//2, 1//3, missing],
+                Union{Missing,Real}[1, 1.5, missing],
+                Union{Missing,Number}[1, 1.5, missing], Any[1, 1.5, missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        @test combine(gdf, :x => fun => :y) ≅ combine(gdf, :x => (x -> fun(x)) => :y)
+    end
+
+    for fun in (first, last, length, first∘skipmissing, last∘skipmissing),
+        col in ([1, 2, 3], [big(1.5), big(2.5), big(3.5)], [1 + 0.5im, 2 + 0.5im, 3 + 0.5im],
+                [true, false, true], [pi, pi, pi], [1//2, 1//3, 1//4],
+                Real[1, 1.5, 1//2], Number[1, 1.5, 1//2], Any[1, 1.5, 1//2],
+                [1, 2, missing], [big(1.5), big(2.5), missing], [1 + 0.5im, 2 + 0.5im, missing],
+                [true, false, missing], [pi, pi, missing], [1//2, 1//3, missing],
+                Union{Missing,Real}[1, 1.5, missing],
+                Union{Missing,Number}[1, 1.5, missing], Any[1, 1.5, missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        if fun isa typeof(last∘skipmissing)
+            # this is another hard corner case
+            if eltype(col) === Any
+                @test_throws MethodError combine(gdf, :x => fun => :y)
+            else
+                @test combine(gdf, :x => fun => :y) ==
+                      combine(groupby_checked(dropmissing(parent(gdf)), :g), :x => fun => :y)
+            end
+            @test_throws MethodError combine(gdf, :x => (x -> fun(x)) => :y)
+        else
+            @test combine(gdf, :x => fun => :y) ≅ combine(gdf, :x => (x -> fun(x)) => :y)
+        end
+    end
+
+    for fun in (sum, mean, var, std),
+        col in ([1:3, 4:6, 7:9], [1:3, 4:6, missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        if eltype(col) >: Missing
+            @test_throws MethodError combine(gdf, :x => fun => :y)
+            @test_throws MethodError combine(gdf, :x => (x -> fun(x)) => :y)
+        else
+            @test combine(gdf, :x => fun => :y) ≅ combine(gdf, :x => (x -> fun(x)) => :y)
+        end
+    end
+
+    for fun in (sum∘skipmissing, mean∘skipmissing),
+        col in ([1:3, 4:6, 7:9], [1:3, 4:6, missing])
+        gdf = groupby_checked(DataFrame(g=[1, 1, 1], x=col), :g)
+        @test combine(gdf, :x => fun => :y) ≅ combine(gdf, :x => (x -> fun(x)) => :y)
+    end
+
+    # workaround https://github.com/JuliaLang/julia/issues/36979
```

Comment:
It's not a workaround, it's just checking that we do the same thing as Base, right?

Reply:
I have changed the comment, but I would keep it. At some point https://github.com/JuliaLang/julia/issues/36979 will get fixed and then these tests will start failing so it is better to understand why then.